<!doctype html>
<html >
<head>
    
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <!--[if lt IE 9]>
                <script src="http://css3-mediaqueries-js.googlecode.com/svn/trunk/css3-mediaqueries.js"></script>
        <![endif]-->
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta http-equiv="Content-Style-Type" content="text/css" />

    <!-- <link rel="stylesheet" type="text/css" href="template.css" /> -->
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/template.css" />

    <link href="https://vjs.zencdn.net/5.4.4/video-js.css" rel="stylesheet" />

    <script src="https://code.jquery.com/jquery-2.2.1.min.js"></script>
    <!-- <script type='text/javascript' src='menu/js/jquery.cookie.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.hoverIntent.minified.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.dcjqaccordion.2.7.min.js'></script> -->

    <!-- <link href="menu/css/skins/blue.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/graphite.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/grey.css" rel="stylesheet" type="text/css" /> -->
  
    <!-- <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> -->
        
  
    <!-- <script src="script.js"></script> -->
  
    <!-- <script src="jquery.sticky-kit.js "></script> -->
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.cookie.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.hoverIntent.minified.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.dcjqaccordion.2.7.min.js'></script>

    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/blue.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/graphite.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/grey.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/ryangrose/easy-pandoc-templates@948e28e5/css/elegant_bootstrap.css" rel="stylesheet" type="text/css" />
  
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/script.js"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/jquery.sticky-kit.js"></script>
    <meta name="generator" content="pandoc" />
  <title>README</title>
  <style type="text/css">code{white-space: pre;}</style>
</head>
<body>

    
    <div class="container">
    <div class="row">
            <div class="span12">

      
      <h1
id="phy-5-a-generator-of-phylogenetic-tree-based-on-genome-wide-comparisons-of-pentamer-profiles.">Phy
5, a generator of phylogenetic tree based on genome-wide comparisons of
pentamer profiles.</h1>
<h3 id="phy5">Phy5</h3>
<p>Phy5 is a Shiny web applicaion, a generator of phylogenetic tree
based on genome-wide comparisons of pentamer profiles. It is written in
R using the Shiny, Biostrings, ape and pvclust libraries.</p>
<p>Phy5 online version is can be used at
http://nakano.no-ip.org:3838/phy5/phy5/ or
https://phy5.shinyapps.io/Phy5R/.</p>
<p>Upload files of fasta-formated DNA sequences, one file for one
strain, and Phy5 calculates pentamaer profiles. The pentanucleotide
profiles can be downloaded as csv files in raw numbers or proportional
numbers, The resultant phylogenetic tree can be downloaded as a pdf
file.</p>
<h3 id="phy5cli">Phy5cli</h3>
<p>Phy5cli is the command line version of Phy5.</p>
<p>Phy5cli is a python script including R function with rpy2 library. It
requires python 3 with the following libraries: numpy, pandas, rpy2, R
(version &gt;=4.0) and its packages: importr, ocalconverter, py2rpy,
rpy2py, pandas2ri, numpy2ri, Biostrings, ape, pvclust, ctc.</p>
<p>Phy5cli is the executable file that works on Linux platform but R and
the packages, Biostrings, ape, and pvclust, are required. The file is
saved in the $PATH and Phy5cli executable without needing to specify the
full path to the file.</p>
<p>Phy5cli can operate on a single FASTA-formatted file, treating each
sequence as a sample.</p>
<p>To run Phy5cli for a single fasta file, run</p>
<p><code>$python3 Phy5cli.py -f fasta.fna -s manhattan -g ward</code></p>
<ul>
<li><code>fasta.fna</code>:Your input single FASTA file.</li>
<li><code>manhattan</code>:Distance method. It should be chosen one out
of ‘euclidean’,‘manhattan’, ‘maximum’, ‘canberra’, and ‘binary’.</li>
<li><code>ward</code>: Agglomeration method. It should be chosen one out
of ‘ward’, ‘average’, ‘single’, ‘complete’, ‘mcquitty’, ‘median’, and
‘centroid’.</li>
</ul>
<p>To run the executable version, run</p>
<p><code>$Phy5cli -f fasta.fna -s manhattan -g ward</code></p>
<p>To run Phy5cli for on a set of FASTA-formatted files, treating each
file as a sample, run</p>
<p><code>$python3 Phy5cli.py -d fastafolder -s manhattan -g ward</code></p>
<ul>
<li><code>fastafolder</code>:The directory containing your input fasta
files.</li>
</ul>
<p>Phy5cli produces two csv tabls of frequencies of penta nucleotides in
each sequence in sigle fasta file or each file in a directory and
proportional values of the frequencies, and the phylogenetic tree whose
file name containing distance method and agglomeration method.</p>
            </div>
    </div>
  </div>
  <script src="https://vjs.zencdn.net/5.4.4/video.js"></script>

</body>
</html>
